geoslurp/discover/fesom/extractprofiles.py

```diff
@@ -32,19 +34,19 @@ def getTemp(fesomURI,tspan,zlvlid):
     #loop over single files 
     for i in range(len(fesomURI)):
         nc_id = Dataset(fesomURI[i]["uri"], 'r')  # open file and obtain its id
-        with nc_id as nc:
-            # extract valid time nodes
-            times=nc_id["time"]
-            units=times.units
-            calendar=times.calendar
-            firstdate=num2date(times[0],units,calendar=calendar)
-            time=sec2month(nc_id)
-            #loop over inside the file
-            for j, t in enumerate(time):
-                if tspan[0] <= t and t < tspan[1]:
-                    temp=np.array(nc_id["temp"][j,zlvlid])
-                    if len(temp_all)==0:
-                        temp_all=[temp]
-                    else: 
-                        temp_all = np.concatenate((temp_all, [temp])) 
+        # with nc_id as nc:
```

Comment:
The `with` statement is actually recommended to open a file, as it will properly clean up and close the open file after exiting the `with` scope. You can try something like:
```
with Dataset(fesomURI[i]["uri"],"r") as nc_id:
            time= ...
```
Otherwise you may have open files dangling around, although in this case the file is going to be implicitly closed once the variable nc_id is reassigned to another Dataset (i.e. the destructor of the old Dataset will be called upon *garbage collection*).